ecl/eclagent/eclagent.cpp

```diff
@@ -2528,6 +2536,13 @@ void EclAgentWorkflowMachine::noteTiming(unsigned wfid, timestamp_type startTime
     const cost_type diskAccessCost = aggregateDiskAccessCost(wu, scope);
     if (diskAccessCost)
         wu->setStatistic(queryStatisticsComponentType(), queryStatisticsComponentName(), SSTworkflow, scope, StCostFileAccess, NULL, diskAccessCost, 1, 0, StatsMergeReplace);
+    stat_type totalSizeSpill = 0, peakSizeSpill = 0;
+    gatherSpillSize(wu, scope, totalSizeSpill, peakSizeSpill);
+    if (totalSizeSpill)
+    {
+        wu->setStatistic(queryStatisticsComponentType(), queryStatisticsComponentName(), SSTworkflow, scope, StSizePeakSpillFile, nullptr, peakSizeSpill, 1, 0, StatsMergeReplace);
+        wu->setStatistic(queryStatisticsComponentType(), queryStatisticsComponentName(), SSTworkflow, scope, StSizeSpillFile, nullptr, totalSizeSpill, 1, 0, StatsMergeReplace);
+    }
```

Comment:
This code is almost identical to the global code above, and could be combined into a function.  This could then be also called from roxie - which needs similar code if it is acting as the eclagent.